projects/fsastorefrontlib/src/cms-components/agent/appointment-scheduling-confirmation/appointment-scheduling-confirmation.component.ts

```diff
@@ -0,0 +1,29 @@
+import { Component, OnInit } from '@angular/core';
+import { DomSanitizer } from '@angular/platform-browser';
+import { RoutingService } from '@spartacus/core';
+import { genericIcons } from '../../../assets/icons/generic-icons';
+
+@Component({
+  selector: 'cx-fs-appointment-scheduling-confirmation',
+  templateUrl: 'appointment-scheduling-confirmation.component.html',
+})
+export class AppointmentSchedulingConfirmationComponent implements OnInit {
+  constructor(
+    protected domSanitizer: DomSanitizer,
+    protected routingService: RoutingService
+  ) {}
+
+  appointmentDate: string;
+
+  ngOnInit() {
+    if (history.state.date) {
+      this.appointmentDate = history.state.date;
```

Comment:
We should definitely avoid using direct reference to browser's API. Not only because we plan to use SSR in future release, but as it's stepping out of Angular application. Can the router state be used here?

Reply:
If we really need browser's history state, then perhaps use Spartecus' `WindowRef` class.
`winRef.nativeWindow.history.state` or something like that. But if it's all possible it would be the best to change implementation of this.